LoopKitUI/OnboardingUI.swift

```diff
@@ -10,6 +10,56 @@ import SwiftUI
 import HealthKit
 import LoopKit
 
+public enum NotificationAuthorization: Int {
+    /// User has not yet made a choice regarding whether the application may schedule or receive user notifications.
+    case notDetermined
+
+    /// User has explicitly denied this application from scheduling or receiving user notifications.
+    case denied
+
+    /// User has authorized this application to schedule or receive non-interruptive user notifications.
+    case provisional
+
+    /// User has authorized this application to schedule or receive any user notifications.
+    case authorized
+}
+
+public protocol NotificationAuthorizationProvider: AnyObject {
+    /// The current notification authorization.
+    ///
+    /// - Parameters:
+    ///     - completion: Invoked when notification authorization is available.
+    func getNotificationAuthorization(_ completion: @escaping (NotificationAuthorization) -> Void)
+
+    /// Authorize notification. Should only be invoked if notificationAuthorization is .notDetermined.
+    ///
+    /// - Parameters:
+    ///     - completion: Invoked when notification authorization is complete along with the resulting authorization.
+    func authorizeNotification(_ completion: @escaping (NotificationAuthorization) -> Void)
+}
+
+public enum HealthStoreAuthorization: Int {
+    /// User has not yet made a choice regarding permissions for one or more health store data types.
+    case notDetermined
+
+    /// User has explicitly chosen permissions for each health store data type.
+    case determined
```

Comment:
[nit] I presume we can't tell which permissions are authorized, nor whether they are read/write?  If we could, would it make sense to represent those here?

Reply:
We could, but it would require a function call/callback for each read and write for each type. We don't ever do anything with the specific permissions given to each type (except deep in the stores), so I don't think we gain anything by doing that extra work here. In this case we really just want to know "has the user been asked for permissions for all of the read/write types we *currently* want". If we ever add a new type (even if the user has previously allowed/denied all of the other types), then this will return `notDetermined` and will show the permissions UI again.

For Bluetooth I think design wants to do something during onboarding that says something like "wait, we just showed you the Bluetooth permissions, but you didn't allow Bluetooth and Loop won't work, so go fix that here". So, we need more information about whether the permissions we granted or denied for that workflow, but for the health store we don't really care what permissions they granted or denied, but just that they made a choice.